### PR TITLE
fix(insights): move 'insightsClient not provided error' to wrapper level

### LINKED ISSUE #4269: The `insightsClient` option has not been provided to `instantsearch`
When I click anywhere on my search result it show a error on console 

**The `insightsClient` option has not been provided to `instantsearch`** 

The same in your netlify example as well - https://instantsearchjs.netlify.com/examples/e-commerce/search/?query=books

**Steps to reproduce**
1. Search anything
2. Click on the search result items

How can I fix this? 

image

### PR DESCRIPTION
**Pull request by tkrugg**
**Summary**

Prior to this change, this error was fired in the click listeners. As a result, users who used a custom component (and therefore a custom listeners) could never see it (they saw instead: insights is `TypeError: insights is not a function`).

Now the insightsClient wrapper itself is responsible for throwing this error.

This also fixes a bug introduced in https://github.com/algolia/instantsearch.js/pull/4197 when the `if (!props.insights) { ... }` was accidentaly brought on top of the handler.

This is not breaking change.

fixes #4269 

**Result**

[View Story →](https://deploy-preview-4254--instantsearchjs.netlify.com/stories/?path=/story/results-hits--with-insights-helper-when-insightsclient-has-not-been-provided)

### CODE REVIEW

File: src/lib/__tests__/insights-client-test.ts [hunk 1]
```diff
@@ -102,7 +123,14 @@ describe('withInsights', () => {
...
 
       const [renderProps] = renderFn.mock.calls[0];
-      expect(renderProps).not.toHaveProperty('insights');
+      expect(() => {
+        renderProps.insights('clickedObjectIDsAfterSearch', {
+          eventName: 'add to favorites',
+          objectIDs: ['1'],
+        });
+      }).toThrowErrorMatchingInlineSnapshot(
+        `"The \`insightsClient\` option has not been provided to \`instantsearch\`."`
```
Comment (eunjae-lee) [inline src/lib/__tests__/insights-client-test.ts]: What if we also provide a url to the doc? Too much?

https://www.algolia.com/doc/guides/getting-insights-and-analytics/search-analytics/click-through-and-conversions/how-to/send-click-and-conversion-events-with-instantsearch/js/

(btw, that's a one long url 😅)

Comment (tkrugg) [reply src/lib/__tests__/insights-client-test.ts]: Good call

Comment (tkrugg) [reply src/lib/__tests__/insights-client-test.ts]: Actually I added a URL to instantsearch docs. it's a generated URL and just as informative.

File: stories/hits.stories.js [hunk 2]
```diff
@@ -140,4 +140,36 @@ storiesOf('Results|Hits', module)
           action(`[InsightsClient] sent ${method} with payload`)(payload),
       }
     )
+  )
+  .add(
+    'with insights helper when insightsClient has not been provided',
```
Comment (francoischalifour) [inline stories/hits.stories.js]: What's the point of this story if it errors and users can't try the feature?

**Pull request merged**